### ISSUE TITLE
Update to 42.0 loses access to existing virtual machines

### ISSUE THREAD

**Issue by neildarlow** [flathub/org.gnome.Boxes#74]
After updating Boxes to 42.0 and libosinfo 1.9 I have lost access to my virtual machine. Running Boxes at the command line reports:

`(gnome-boxes:2): Gtk-WARNING **: 10:34:40.173: GtkFlowBox with a model will ignore sort and filter functions

(gnome-boxes:2): Gtk-WARNING **: 10:34:40.174: GtkListBox with a model will ignore sort and filter functions
Error setting up default broker: Failed to start storage pool: cannot open directory '/home/neil/.var/app/org.gnome.Boxes/data/gnome-boxes/images': No such file or directory

(gnome-boxes:2): Boxes-WARNING **: 10:34:40.597: (../src/app.vala:443):boxes_app_setup_default_source_co: runtime check failed: (default_connection != null)`

The directory indicated does exist in my home directory and has sifficient read access e.g.:

`[neil@vivobook ~]$ ls -lR .var/app/org.gnome.Boxes/data
.var/app/org.gnome.Boxes/data:
total 4
drwxr-xr-x. 1 neil neil  12 Apr  5  2021 gnome-boxes
drwxr-xr-x. 1 neil neil   0 May  2  2021 keyrings
-rw-r--r--. 1 neil neil 218 Feb 25 13:28 recently-used.xbel

.var/app/org.gnome.Boxes/data/gnome-boxes:
total 0
drwxr-xr-x. 1 neil neil 22 May  5  2021 images

.var/app/org.gnome.Boxes/data/gnome-boxes/images:
total 7262016
-rw-r--r--. 1 neil neil 7436304384 Feb 25 13:28 freebsd12.2

.var/app/org.gnome.Boxes/data/keyrings:
total 0`

**Comment by felipeborges** [comment]
Is there another instance of GNOME Boxes stuck in the background?

`flatpak ps`

and `flatpak kill org.gnome.Boxes`

**Comment by neildarlow** [comment]
No. The issue is evident after a fresh boot and login. The Boxes UI is shown and you can quit the application but then an instance is left running and flatpak ps shows this.

**Comment by felipeborges** [comment]
I noticed some issues with qemu locking the backing file and the xdg-desktop-portal document portal. By enabling --filesystem=host this issues seem to go away. I sent a flathub update to poke that whole in the sandbox for now until I can find a proper solution that works down the stack. Please, let me know if that fixes your issue too (after updating).

**Comment by neildarlow** [comment]
Sent with Geary Mail on fedora 35 Silverblue edition.
Hi Felipe,

More complex than I thought. I did have a concern that the issue might 
be related to home directories on Silverblue being under /var/home by 
default. For Silverblue, filesystem=host probably isn't a great concern 
but it could be for less hardened distributions.

Thank you for your fast response on this.
